### ISSUE TITLE
Player movement is not 100% smooth

### ISSUE THREAD

**Issue by dRaiNe** [vmangos/core#129]
**Current behavior**: 
See https://youtu.be/vBn_W8ppQOw?t=3 at 0:04. Player's position bounces slightly from time to time.
**Expected behavior**: 
Players should move smoothly.
**Steps to reproduce**:

1. Open 2 WoW Clients at the same time and log into 2 different accounts.
2. Let first player stand still and second runs around like in the video above. Use windowed mode.
3. Observe stuttering on localhost.

**Client version**: 
1.12
**Commit hash**: 

**Operating system**: 
Windows 7 x64

**Comment by ratkosrb** [comment]
The other servers in that video are not for vanilla, so its possible this is a flaw of the vanilla client.

**Comment by dRaiNe** [comment]
No, it's not. I experienced this also with WotLK trinitycore some time ago. Notice all the 3 servers in the video have this problem. The Felmyst does it and the Warmane (TBC) seems ok in the video, but I tested on their server and believe they have the same issue, only the update diff is somewhat lower (so it's harder to spot).

For a split of a second, the player blinks from one location to another and then back immediately. See for example https://www.youtube.com/watch?v=VX7RuUcd-3Q&t=588s a blizz vanilla rogue. I don't believe he has the best latency all the time, yet player movements are smooth and this 'teleporting' never happened on the retail.

Also, it's only player movement problem. NPCs move smoothly all the time, no stuttering!

**Comment by Daribon** [comment]
Can confirm, it's a serverside issue that exist on all vanilla and TBC emulators.

**Comment by Kittnz** [comment]
https://github.com/TrinityCore/TrinityCore/pull/18771

**Comment by ratkosrb** [comment]
For some reason, movement appears smoother on the 1.8.4 client.
https://streamable.com/qppp5

Now compare to the constant twitching with 1.12.
https://streamable.com/k9qvb

**Comment by ax-ze** [comment]
Is this still being looked at? The movement is the only thing that is still immersion breaking.

**Comment by Kittnz** [comment]
I guess different movement packets @ratkosrb

**Comment by Test045** [comment]
The twitching only happens when the player moves while also moving his camera.
One should see how we handle MSG_MOVE_SET_FACING (when already moving only)

EDIT : It's actually reproductible even with "turn camera" keybind (which sends a different packet  than MSG_MOVE_SET_FACING), it's just harder to reproduce in that case for some reason.

**Comment by ulfgebhardt** [comment]
CMangos works on some movement packet parsing at the moment. Mind take a look @ratkosrb  ?

**Comment by ratkosrb** [comment]
Cmangos does not have this solved.
